### ISSUE TITLE
how to solve the repetition issues when finetune large model

### ISSUE THREAD

**Issue by andysun-sw** [facebookresearch/ParlAI#4374]
I use the following way to finetune reddit model:
parlai train_model -t blended_skill_talk  -veps 0.25 --attention-dropout 0.0 --batchsize 16 --model transformer/generator --embedding-size 2560 --ffn-size 10240 --variant prelayernorm --n-heads 32 --n-positions 128 --n-encoder-layers 2 --n-decoder-layers 24 --history-add-global-end-token end --delimiter '  ' --dict-tokenizer bytelevelbpe  --dropout 0.1 --fp16 True --init-model zoo:blender/reddit_3B/model --dict-file zoo:blender/reddit_3B/model.dict --label-truncate 128 --log_every_n_secs 10 -lr 7e-06 --lr-scheduler reduceonplateau --lr-scheduler-patience 3 --optimizer adam --relu-dropout 0.0 --activation gelu --model-parallel true --save-after-valid True --text-truncate 128 --truncate 128 --warmup_updates 100 --fp16-impl mem_efficient --update-freq 2 --gradient-clip 0.1 --skip-generation True -vp 4 -vmt ppl -vmm min --model-file /py_torch/parlai_finetune_model/test_train_27B

but the system always generate same thing repeatly:
![Image  2](https://user-images.githubusercontent.com/89009298/155109931-d2ed2a0c-a7ab-4f89-894a-4aaca98abee3.png)

![Image](https://user-images.githubusercontent.com/89009298/155110214-d7d1a66e-8eea-4415-8e8a-47ace4e59190.png)

Are there some parameters I can adjust to solve this problem?

Or I need to change loss function or use the method like Exposure Bias Solutions? If I should use this way, do you have some suggestions about which codes could change at first?

**Comment by klshuster** [comment]
repetition is an open problem. you can try setting some generation settings to enforce beam blocking (e.g. `--beam-block-ngram 3`), or you can try training with [unlikelihood](https://github.com/facebookresearch/ParlAI/tree/main/projects/dialogue_unlikelihood)
